backend/app/services/subscription/notification_dispatcher.py

```
@@ -910,6 +1165,55 @@ async def _send_custom_webhook(
+
+        # Find all attachment URLs and replace with public share URLs
+        def replace_url(match: re.Match) -> str:
+            attachment_id = int(match.group(1))
+
+            # Generate public share token for this attachment
+            try:
+                token = _generate_public_share_token(
+                    attachment_id=attachment_id, expires_in_days=7
+                )
+                return f"{base_url}/download/shared?token={token}"
+            except Exception as e:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)